### PR TITLE
Made autoscale able to scale down below 9

### PR DESCRIPTION
**Pull request by mjksmith**
(9 instances)

Also made PID autoscaling able to scale as fast as it thinks is correct

New autoscaling math/voodoo is based off of [this paper](https://users.cs.duke.edu/~shivnath/papers/acdc09.pdf)

### CODE REVIEW

**Comment by mjksmith** [comment]
(also this fixes a couple giant bugs in autoscaling that currently cause it not to work at all)

File: paasta_tools/autoscaling_lib.py [hunk 1]
```diff
@@ -85,36 +84,38 @@ class MetricsProviderNoDataError(ValueError):
 
 
 @register_autoscaling_component('threshold', DECISION_POLICY_KEY)
-def threshold_decision_policy(marathon_service_config, error, threshold=0.1, **kwargs):
+def threshold_decision_policy(marathon_service_config, current_instances, error, **kwargs):
     """
-    Decides to autoscale a service up or down if the service utilization exceeds the setpoint
-    by a certain threshold.
+    Desides to autoscale up or down by 10% if the service exceeds the upper or lower thresholds
```
Comment (Rob-Johnson) [inline paasta_tools/autoscaling_lib.py]: s/Desides/Decides

**Comment by mjksmith** [comment]
Added test coverage for the big autoscaling bugs.

**Comment by solarkennedy** [comment]
ship

**Pull request merged**